ktor-features/ktor-webjars/src/io/ktor/webjars/Webjars.kt

```diff
@@ -0,0 +1,85 @@
+package io.ktor.webjars
+
+import io.ktor.application.ApplicationCall
+import io.ktor.application.ApplicationCallPipeline
+import io.ktor.application.ApplicationFeature
+import io.ktor.application.call
+import io.ktor.http.ContentType
+import io.ktor.http.HttpMethod
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.defaultForFilePath
+import io.ktor.pipeline.PipelineContext
+import io.ktor.request.httpMethod
+import io.ktor.request.uri
+import io.ktor.response.respond
+import io.ktor.response.respondBytes
+import io.ktor.util.AttributeKey
+import org.webjars.MultipleMatchesException
+import org.webjars.WebJarAssetLocator
+import java.nio.file.Paths
+
+class Webjars(val configuration: Configuration) {
+
+    private fun fileName(path: String) : String = Paths.get(path).fileName.toString()
+
+    private fun extractWebJar(path : String) : String {
+        val firstSlash = if (path.startsWith("/")) 1 else 0
+        val nextSlash = path.indexOf("/", 1)
+        val webjar = if(nextSlash > -1) path.substring(firstSlash,  nextSlash) else ""
+        val partialPath = path.substring(nextSlash + 1)
```

Comment:
`"some-path"` -> `webjar` = `""` and `partialPath` = `"some-path"` Looks strange.. Shouldn't we simply return immediately (for example, `null` and then by-pass it or respond 404)

Reply:
Thanks I'm really bad in naming things. So `rootPath` is the configuration.path you set (for instance `/webjars/`), `resourcePath` is whatever is left after that (`/jquery/js/jquery.js`), the `webjar` is the first part of that path, could be `jquery`, `bootstrap` ... and finally the `partialPath` is `/js/jquery.js` I hope this makes thing clearer